### PR TITLE
New code to detect pauses in GB-580P track

### PR DESCRIPTION
**Pull request by xeonvs**
I'm added new method to detect pauses in GB-580P by Speed & Cadence track

### CODE REVIEW

File: Import/ImportJob_Globalsat.cs [hunk 1]
```diff
@@ -264,114 +268,125 @@ protected void AddActivities(IImportResults importResults, IList<GlobalsatPacket
...
-                        bool insertPause = false;
-                        if (activity.GPSRoute.Count > 0)
-                        {
-                            float gpsDist = gpsPoint.DistanceMetersToPoint(activity.GPSRoute[activity.GPSRoute.Count - 1].Value);
-                            //Some limit on when to include pause
-                            //gpsDist must be higher than (all) GPS errors
-                            if (gpsDist > 100 && gpsDist > 3 * dist)
+                        
+                        //speed & cadence method
+                        if (detectPauses == 2 && activity.CadencePerMinuteTrack != null)
```
Comment (gerhardol) [inline Import/ImportJob_Globalsat.cs]: Pause check could be detected also without GPS using the speed track.
A reason to implement this is though that my 580P GPS is broken so I cannot test this as is.

File: Data/DeviceConfigurationInfo.cs [hunk 2]
```diff
@@ -134,6 +140,7 @@ private static class xmlTags
             public const string AllowedIdsText = "allowedids";
             public const string ImportSpeedDistanceTrack = "ImportSpeedDistanceTrack";
             public const string DetectPausesFromSpeedTrack = "DetectPausesFromSpeedTrack";
```
Comment (gerhardol) [inline Data/DeviceConfigurationInfo.cs]: Add a comment that this is a legacy parameter

Comment (xeonvs) [inline Data/DeviceConfigurationInfo.cs]: ok, thanks.

File: UI/DeviceConfigurationDlg.cs [hunk 3]
```diff
@@ -59,7 +59,9 @@ public DeviceConfigurationDlg(FitnessDevice_Globalsat fitnessDevice, bool detect
             btnCancel.Click += new EventHandler(btnCancel_Click);
             this.chkImportOnlyNew.Checked = this.fitnessDevice.configInfo.ImportOnlyNew;
             this.chkImportDistance.Checked = this.fitnessDevice.configInfo.ImportSpeedDistanceTrack;
-            this.chkDetectPausesFromSpeed.Checked = this.fitnessDevice.configInfo.DetectPausesFromSpeedTrack;
+            this.optDetectPausesFromSpeed.Checked = this.fitnessDevice.configInfo.DetectPauses == 1 ? true : false;
+            this.optDetectPausesFromSpeedCadence.Checked = this.fitnessDevice.configInfo.DetectPauses == 2 ? true : false;
+            this.optDetectPausesOff.Checked = this.fitnessDevice.configInfo.DetectPauses == 0 ? true : false;
```
Comment (gerhardol) [inline UI/DeviceConfigurationDlg.cs]: Set condition to be active if none other is active - easier to maintain

File: UI/DeviceConfigurationDlg.cs [hunk 4]
```diff
@@ -126,7 +128,7 @@ internal DeviceConfigurationInfo ConfigurationInfo
             {
                 this.fitnessDevice.configInfo.ImportOnlyNew = this.chkImportOnlyNew.Checked;
                 this.fitnessDevice.configInfo.ImportSpeedDistanceTrack = this.chkImportDistance.Checked;
-                this.fitnessDevice.configInfo.DetectPausesFromSpeedTrack = this.chkDetectPausesFromSpeed.Checked;
+                //this.fitnessDevice.configInfo.DetectPauses = this.optDetectPausesFromSpeed.Checked;
```
Comment (gerhardol) [inline UI/DeviceConfigurationDlg.cs]: Is not this required?

File: Import/ImportJob_Globalsat.cs (hunk 1 above)
Comment (xeonvs) [inline Import/ImportJob_Globalsat.cs]: I added a new code in addition to the old one and changed settings form for selecting it.

File: UI/DeviceConfigurationDlg.cs (hunk 4 above)
Comment (xeonvs) [inline UI/DeviceConfigurationDlg.cs]: yep, i used variable

File: UI/DeviceConfigurationDlg.Designer.cs [hunk 5]
```diff
@@ -115,14 +119,26 @@ private void InitializeComponent()
             this.btnCancel.TextLeftMargin = 2;
             this.btnCancel.TextRightMargin = 2;
             // 
+            // chkImportDistance
```
Comment (gerhardol) [inline UI/DeviceConfigurationDlg.Designer.cs]: Why was this removed?

File: Import/ImportJob_Globalsat_fixspeed.cs [hunk 6]
```diff
@@ -0,0 +1,546 @@
+/*
```
Comment (gerhardol) [inline Import/ImportJob_Globalsat_fixspeed.cs]: Is this file really used?

File: UI/DeviceConfigurationDlg.Designer.cs (hunk 5 above)
Comment (xeonvs) [inline UI/DeviceConfigurationDlg.Designer.cs]: code designer change block position

File: Import/ImportJob_Globalsat_fixspeed.cs (hunk 6 above)
Comment (xeonvs) [inline Import/ImportJob_Globalsat_fixspeed.cs]: no, its backup.

File: UI/DeviceConfigurationDlg.cs [hunk 7]
```diff
@@ -59,7 +59,21 @@ public DeviceConfigurationDlg(FitnessDevice_Globalsat fitnessDevice, bool detect
             btnCancel.Click += new EventHandler(btnCancel_Click);
             this.chkImportOnlyNew.Checked = this.fitnessDevice.configInfo.ImportOnlyNew;
             this.chkImportDistance.Checked = this.fitnessDevice.configInfo.ImportSpeedDistanceTrack;
-            this.chkDetectPausesFromSpeed.Checked = this.fitnessDevice.configInfo.DetectPausesFromSpeedTrack;
+            switch (this.fitnessDevice.configInfo.DetectPauses)
+            {
+                case 0:
```
Comment (gerhardol) [inline UI/DeviceConfigurationDlg.cs]: default: should act on this too (forward compatibility)

File: UI/DeviceConfigurationDlg.cs (hunk 4 above)
Comment (gerhardol) [inline UI/DeviceConfigurationDlg.cs]: Add a comment here, noting that the value is read from defaults, updated at changes

File: UI/DeviceConfigurationDlg.cs (hunk 7 above)
Comment (xeonvs) [inline UI/DeviceConfigurationDlg.cs]: fixed

**Comment by gerhardol** [comment]
A few more comments. No testing done, will not do much. 
Do you feel that this is ready for release?

**Comment by xeonvs** [comment]
> Do you feel that this is ready for release?
I think need additional testing

File: UI/DeviceConfigurationDlg.cs [hunk 8]
```diff
@@ -140,7 +143,7 @@ internal DeviceConfigurationInfo ConfigurationInfo
             {
                 this.fitnessDevice.configInfo.ImportOnlyNew = this.chkImportOnlyNew.Checked;
                 this.fitnessDevice.configInfo.ImportSpeedDistanceTrack = this.chkImportDistance.Checked;
-                //this.fitnessDevice.configInfo.DetectPauses = this.optDetectPausesFromSpeed.Checked;
+                this.fitnessDevice.configInfo.DetectPauses = 0;
```
Comment (gerhardol) [inline UI/DeviceConfigurationDlg.cs]: This will overwrite the changes in the GUI...

**Comment by gerhardol** [comment]
I will merge locally, make a few changes, apply another unrelated change, then push. avoid making more changes in this branch

File: UI/DeviceConfigurationDlg.cs (hunk 8 above)
Comment (xeonvs) [inline UI/DeviceConfigurationDlg.cs]: Hmm... It's default block? Now, GUI not changed.

**Comment by xeonvs** [comment]
ok

File: UI/DeviceConfigurationDlg.cs (hunk 8 above)
Comment (xeonvs) [inline UI/DeviceConfigurationDlg.cs]: Tested, you're right. This change broke GUI.

**Comment by gerhardol** [comment]
Update translation too

**Comment by gerhardol** [comment]
Testing done?

**Comment by xeonvs** [comment]
Testing isn't finished yet. So far there is still a problem with
synchronization of time in lap.

2017-04-29 1:10 GMT+03:00 Gerhard Olsson <notifications@github.com>:
